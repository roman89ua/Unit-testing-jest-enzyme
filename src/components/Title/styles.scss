// <div className="title">
//       <div className="divider" />
//       <span>{title}</span>
//     </div>

.title {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  .divider {
    background-color: dodgerblue;
    padding: 0 0 3px 0;
    width: 80px;
  }

  .title-text {
    color: dimgray;
  }
}
