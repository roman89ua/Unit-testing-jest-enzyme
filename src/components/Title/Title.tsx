import React from 'react';
import './styles.scss';

interface ITitle {
  title: string | null | undefined;
}

export const Title = ({ title }: ITitle) => {
  if (!title) {
    return null;
  }
  return (
    <div className="title">
      <div className="divider" />
      <h1 className="title-text">{title}</h1>
    </div>
  );
};
