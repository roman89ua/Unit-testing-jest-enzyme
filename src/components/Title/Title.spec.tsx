import React from 'react';
import { ReactWrapper, ShallowWrapper, mount, shallow } from 'enzyme';
import { Title } from './Title';

describe('<Title /> component shalow rendering', () => {
  const APP_TITLE = 'App Title.';
  let container: ShallowWrapper;

  beforeEach(() => {
    container = shallow(<Title title={APP_TITLE} />);
  });

  it('should render one header', () => {
    const header = container?.find('.title-text');
    expect(header).toHaveLength(1);
  });

  it(`should render ${APP_TITLE} text`, () => {
    expect(container).toMatchSnapshot();
  });
});

describe('<Title /> component full DOM rendering', () => {
  let container: ReactWrapper;

  afterEach(() => {
    container.unmount();
  });

  it('should not render any Title component', () => {
    container = mount(<Title title={undefined} />);
    expect(container?.find('.title').exists()).toBeFalsy();
  });
});
