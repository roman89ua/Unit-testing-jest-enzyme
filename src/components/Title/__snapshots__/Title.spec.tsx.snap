// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> component shalow rendering should render App Title. text 1`] = `
<div
  className="title"
>
  <div
    className="divider"
  />
  <h1
    className="title-text"
  >
    App Title.
  </h1>
</div>
`;
